### ISSUE TITLE
CDK DataSource needs extensive documentation

### ISSUE THREAD

**Issue by Maximaximum** [angular/material2#8107]
It would be great to have a documentation page for a cdk DataSource.

Currently I'm struggling to make use of it in a simple MatList, and I can't find out how to do it. So, when I decide to use a list instead of a table, I have to rewrite my typescript code so that my component uses a plain collection instead of a DataSource.

**Comment by jelbourn** [comment]
The data source just emits the list of items. It's up the the consuming component to decide what to do with that.

**Comment by Maximaximum** [comment]
Yes, but I have tried to use it in an *ngFor and did not succeed. I think documentation would be really helpful

**Comment by willshowell** [comment]
@Maximaximum 

```html
<div *ngFor="let item of items$ | async"> ... </div>
```

```ts
ngOnInit() {
  items$ = this.dataSource.connect();
}

ngOnDestroy() {
  this.dataSource.disconnect();
}
```

**Comment by Maximaximum** [comment]
Ok, I've taken a look at the source code and found out that the `DataSource` class is extremely simple and it has only 2 methods. I'm not sure why, but trying to bind to `dataSource.connect()` didn't work for me when I was opening this issue. But I think that is no longer relevant because of the simplicity of the class.

Anyway, now I think that it would be nice to have a little mention here https://material.angular.io/cdk/table/overview saying that `DataSource` is actually a very simple abstract class, and **to create your own DataSource you actually only need an object having two methods: connect() and disconnect()**. Otherwise, lots of people would have to dig into the source code to understand this very simple yet important information.

**Comment by michaeljota** [comment]
If DataSource is an abstract class, that only have two methods that needs to be implemented, shouldn't be better to expose it as an interface? I know a class can be implemented in Typescript, but I think that if that's all this need, maybe it's better to expose it that way.

And, does the consumer of the DataSource expect a data property in the instance? If so, that I think it is, then this should be also declared in the interface, with type being extends an array of `T`, or a promise or observable that resolves of something of that same type.

If not, then I think any way this should be something that benefice the use of DataSource, as it would have an opinion about where the data should be, and the shape of that data.

**Comment by crisbeto** [comment]
We're in the process of cleaning up our issue backlog. I'm closing this issue, because I don't see what there is to document about the data source. You can implement one yourself by extending the abstract class and then it's up to the consuming component to use it.
